### PR TITLE
add leaky_relu op

### PR DESCRIPTION
**Pull request by beverlylytle**
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

Adds the leaky_relu op.

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: thunder/torch/__init__.py [hunk 1]
```diff
@@ -1801,6 +1801,17 @@ def gelu(a: TensorProxy, /, *, approximate: str = "none") -> TensorLike:
         raise ValueError(f"gelu does not support the approximate={approximate} argument")
 
 
+@torchsymbol(torch.nn.functional.leaky_relu, is_method=False)
+def leaky_relu(a: TensorProxy, /, negative_slope=0.01, inplace: bool = False) -> TensorLike:
```
Comment (t-vi) [inline thunder/torch/__init__.py:1805]: Would it make sense to include all parameters?
 
https://pytorch.org/docs/stable/generated/torch.ao.nn.quantized.functional.leaky_relu.html#torch.ao.nn.quantized.functional.leaky_relu

File: thunder/tests/opinfos.py [hunk 2]
```diff
@@ -1665,6 +1669,18 @@ def elementwise_unary_with_alpha_generator(op, device, dtype, requires_grad):
 elementwise_unary_ops.append(elu_opinfo)
 
 
+leaky_relu_opinfo = OpInfo(
```
Comment (t-vi) [inline thunder/tests/opinfos.py:1672]: Great job adding opinfos!

**Review by t-vi — COMMENTED** [review]
Supercool, thank you @beverlylytle

File: thunder/torch/__init__.py (hunk 1 above)
Comment (IvanYashchuk) [reply thunder/torch/__init__.py:1805]: The link is for `torch.ao.nn.quantized.functional.leaky_relu`. Docs search in PyTorch is not great with many similar results. Do you suggest supporting the `torch.ao.nn.quantized.functional` namespace? Or is it just the confusion with links?
Here's the link for the `torch.nn.functional.leaky_relu` function that is implemented here: https://pytorch.org/docs/stable/generated/torch.nn.functional.leaky_relu.html#torch.nn.functional.leaky_relu

Comment (beverlylytle) [reply thunder/torch/__init__.py:1805]: I was following https://pytorch.org/docs/stable/generated/torch.nn.functional.leaky_relu.html#torch.nn.functional.leaky_relu 

What are the plans to support torch.ao.nn.quantized.functional?

Comment (t-vi) [reply thunder/torch/__init__.py:1805]: Yeah, I'm stupid, sorry.  :)

File: thunder/tests/opinfos.py [hunk 3]
```diff
@@ -1665,6 +1669,18 @@ def elementwise_unary_with_alpha_generator(op, device, dtype, requires_grad):
...
+    ltorch.leaky_relu,
+    dtypes=(datatypes.floating,),
+    sample_input_generator=get_elementwise_unary_with_kwargs_generator([{}, {"negative_slope": 0.5}]),
```
Comment (mruberry) [inline thunder/tests/opinfos.py:1675]: Nice refactor of the sample input generators to accommodate different kwargs

File: thunder/tests/opinfos.py [hunk 4]
```diff
@@ -1665,6 +1669,18 @@ def elementwise_unary_with_alpha_generator(op, device, dtype, requires_grad):
...
+    torch_reference=torch.nn.functional.leaky_relu,
+    # fdm.jvp, which is used in test_vjp_correctness, behaves badly on (-1e-6, 1e-6) for this function
```
Comment (mruberry) [inline thunder/tests/opinfos.py:1677]: Nice comment

File: thunder/torch/__init__.py (hunk 1 above)
Comment (mruberry) [inline thunder/torch/__init__.py:1805]: Nit: `negative_slope: Number = 0.01` or `negative_slope: float` if `negative_slope` has to be a float

**Review by mruberry — APPROVED** [review]
Test failures are unrelated. Fyi @t-vi re: ci issues

File: thunder/torch/__init__.py (hunk 1 above)
Comment (beverlylytle) [reply thunder/torch/__init__.py]: Thanks, should be a float.

**Pull request merged**